src/FragmentCache.php

```diff
@@ -93,11 +99,11 @@ public function start(): void
      */
     public function run(): string
     {
-        if (($content = $this->getCachedContent()) !== false) {
+        if (($content = $this->getCachedContent()) !== null) {
             return $content;
         }
 
-        if ($this->cache instanceof CacheInterface) {
+        if ($this->cache instanceof CacheInterface && $this->noCache === false) {
```

Comment:
Both checks are redundant. There's no sense in having no cache and `$this->cache` is always defined and an instance of `CacheInterface`. You can pass a dummy cache here if needed to turn off caching.

Reply:
fixed.